libnavigation-copilot/src/main/java/com/mapbox/navigation/copilot/MapboxCopilotImpl.kt

```diff
@@ -172,6 +178,7 @@ internal class MapboxCopilotImpl(
     fun stop() {
         unregisterUserFeedbackCallback(userFeedbackCallback)
         appLifecycleOwner.lifecycle.removeObserver(foregroundBackgroundLifecycleObserver)
+        appLifecycleOwnerCleanupAction()
```

Comment:
Let's also nullify it.

Reply:
`appLifecycleOwnerCleanupAction` is val and created once per `MapboxCopilotInstance` to cleanup resources acquired in constructor.

Reply:
@dzinad , do I understand your worries correctly: somebody can call internal method `onStop` a few times, which is not currently possible via public API, so that listener of activity lifecycle will be unregistered twice, where second unregistration does nothing?
I don't see any dangerous consequences with current approach. Can you please help me understand what exactly can go wrong there?

Reply:
I have a little bit another concern here. In case if `MapboxNavigationApp.isSetup()` is false, we call `attachAllActivities` only on `MapboxCopilotImpl` initialization, but call `detachAllActivities` when `MapboxCopilotImpl.stop()` is called. So there should be assumption that `stop` can be called only once. However, we're not protected from api misuse (even if it's internal class).

What if we will be assigning `appLifecycleOwnerCleanupAction` on `start()` call, and assign null on `stop()`?

Reply:
> What if we will be assigning appLifecycleOwnerCleanupAction on start() call, and assign null on stop()?

In that case a different misuse is possible. If onStart is called twice, one of cleanup actions will be lost.

I don't think it's worth putting effort in protecting internal API from misuse if the price is code complexity.

Reply:
> so that listener of activity lifecycle will be unregistered twice, where second unregistration does nothing?

It currently doesn't do nothing, it detaches all activities. It may result in being nothing, but I don't see any strong reason why we should rely on that. We can also add something else to `appLifecycleOwnerCleanupAction`, which won't be a no-op.

Ideally I'd move the setup to `start` and nullify it in `stop` (sth like what Dima suggested). And also add a check that start/stop is not invoked twice in a row (make the second invocation no-op).
But I don't find this critical. If we do that, we will reduce the risk of making an error in the future. But it's up to you whether you find this risk significant.

Reply:
> Ideally I'd move the setup to start and nullify it in stop (sth like what Dima suggested). And also add a check that start/stop is not invoked twice in a row (make the second invocation no-op).

It's already implemented in `MapboxCopilot`. `MapboxCopilot` won't let `MapboxCopilotImpl#onStart` to be called 2 times. `MapboxCopilotImpl` could have been a private class if not unit tests which use it directly.